core/application/impl/app_config_impl.cpp

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/app_config_impl.hpp"
+
+#include <rapidjson/document.h>
+#include <rapidjson/filereadstream.h>
+#include <boost/program_options.hpp>
+#include <iostream>
+
+namespace {
+  struct RegisteredHandlers {
+    using Handler =
+        void (kagome::application::AppConfigurationImpl::*)(rapidjson::Value &);
+    char const *segment_name;
+    Handler handler;
+  };
+
+  template <typename T, typename Func>
+  inline void find_argument(boost::program_options::variables_map &vm,
+                            char const *name,
+                            Func &&f) {
+    assert(nullptr != name);
+    auto it = vm.find(name);
+    if (it != vm.end()) std::forward<Func>(f)(it->second.as<T>());
```

Comment:
I guess forward redundant here. Explain this using, pls.

Reply:
https://stackoverflow.com/questions/31253334/when-should-i-stdforward-a-function-call

Reply:
Now I'm sure it's useless right here. The function gets functor by rvalue, and one of the reasons for this is way to automatically use the special ref-qualified method. But this is not templated function, and the functor becomes an lvalue (because it got the name), so method for lvalue will be called everywhere: `void operator()(...) & {...}`

Reply:
It's very strange, because it seems to be necessary right here) This method expects something with operator(). So in the case if it will be a functor and without ```forward``` ```operator()() &&``` will never be called. With forwarding, the called function version depends on ref-qualifier. :)

Reply:
I was wrong, sorry. It usefil here because this is template (I missed this).